core/src/main/scala/org/apache/spark/status/AppStatusStore.scala

```diff
@@ -421,6 +420,51 @@ private[spark] class AppStatusStore(
     constructTaskDataList(taskDataWrapperIter)
   }
 
+  def exceptionSummary(stageId: Int, attemptId: Int): Seq[v1.ExceptionSummary] = {
+    val (stageData, _) = stageAttempt(stageId, attemptId)
+    val key = Array(stageId, attemptId, stageData.numFailedTasks)
+    asOption(store.read(classOf[CachedExceptionSummary], key))
+      .map(_.exceptionSummary)
+      .getOrElse {
+        val exceptionSummary = computeExceptionSummary(stageId, attemptId)
+        val cachedExceptionSummary = new CachedExceptionSummary(
+          stageId,
+          attemptId,
+          stageData.numFailedTasks,
+          exceptionSummary
+        )
+        store.write(cachedExceptionSummary)
```

Comment:
`CachedExceptionSummary` seems to never be deleted. This would be a potential cause of memory leak?